PeerAddress: divorce from `Message`

schildbach opened the pull request:
It is never sent on its own, so it doesn't need to be a `Message`.

* Static constructur `read(protocolVariant)` replaces the native constructor that deserialized from a payload.
* `write()` helper replaces `bitcoinSerializeToStream()`.
* `serialize()` and `getMessageSize()` helpers replace `bitcoinSerialize()`.

schildbach commented:
Best to review #3015 and #3016 first.

msgilligan reviewed (approved):
LGTM. One minor change to a comment recommended.

schildbach commented on core/src/main/java/org/bitcoinj/core/PeerAddress.java:
```diff
@@ -83,113 +81,180 @@ static Optional<NetworkId> of(int value) {
-     * @param payload Bitcoin protocol formatted byte array containing message content.
-     * @param serializer the serializer to use for this message.
-     * @throws ProtocolException
+     * Deserialize this perr address from a given payload, using a given protocol variant. The variant can be 0
```
Fixed, thanks!

schildbach commented:
I rebased this, so there are some changes:

* Protocol variant 0 doesn't exist any more, after cc5d735eb4bf8146f6153b0603df860b982476a2 was merged.
* I renamed the static constructors to `simple()` (mostly for tests) and `inet()`. Future constructors could be named `onion()` and `i2p()`, but I suspect at that point we might introduce a class hierarchy…

I'm not sure if you want to re-review this.

schildbach commented:
Merged.
